pkg/server/drain_test.go

```diff
@@ -204,3 +208,241 @@ func getAdminClientForServer(
 	client := serverpb.NewAdminClient(conn)
 	return client, func() { _ = conn.Close() }, nil
 }
+
+// TestDrainIgnoresExpiredLeases checks that the drain process
+// does not linger forever in the presence of expired leases.
+func TestDrainIgnoresExpiredLeases(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+
+	stickyEngineRegistry := server.NewStickyInMemEnginesRegistry()
+	defer stickyEngineRegistry.CloseAllStickyInMemEngines()
+	const numServers int = 4
+	stickyServerArgs := make(map[int]base.TestServerArgs)
+	for i := 0; i < numServers; i++ {
+		stickyServerArgs[i] = base.TestServerArgs{
+			Locality: roachpb.Locality{Tiers: []roachpb.Tier{{Key: "node", Value: "n" + strconv.FormatInt(int64(i+1), 10)}}},
+			StoreSpecs: []base.StoreSpec{
+				{
+					InMemory:               true,
+					StickyInMemoryEngineID: t.Name() + strconv.FormatInt(int64(i), 10),
+				},
+			},
+			Knobs: base.TestingKnobs{
+				Server: &server.TestingKnobs{
+					StickyEngineRegistry: stickyEngineRegistry,
+				},
+			},
+		}
+	}
+
+	tc := testcluster.StartTestCluster(t, 4, base.TestClusterArgs{
+		ReplicationMode:   base.ReplicationAuto,
+		ServerArgsPerNode: stickyServerArgs,
+	})
+	defer tc.Stopper().Stop(ctx)
+
+	t.Logf("setting up data...")
+	s0 := tc.Server(0).(*server.TestServer)
+	if err := s0.RunLocalSQL(ctx, func(ctx context.Context, ie *sql.InternalExecutor) error {
+		for _, stmt := range []string{
+			// Adjust the zone configs so that data spreads over the storage
+			// nodes, excluding the observer node.
```

Comment:
Is this setup necessary? Looking at this test I'm not sure why you need/want a dedicated observer node. Shouldn't this all work if you set up a three node cluster with manual replication, and run a request, and drain the node that ends up getting the lease?